app/models/import.rb

```diff
@@ -177,6 +185,39 @@ def parsed_csv
 
     def sanitize_number(value)
       return "" if value.nil?
-      value.gsub(/[^\d.\-]/, "")
+
+      format = NUMBER_FORMATS[number_format]
+      return "" unless format
+
+      # First, normalize spaces and remove any characters that aren't numbers, delimiters, separators, or minus signs
+      sanitized = value.to_s.strip
+
+      # Handle French/Scandinavian format specially
+      if format[:delimiter] == " "
+        sanitized = sanitized.gsub(/\s+/, "") # Remove all spaces first
+      else
+        sanitized = sanitized.gsub(/[^\d#{Regexp.escape(format[:delimiter])}#{Regexp.escape(format[:separator])}\-]/, "")
+
+        # Replace delimiter with empty string
+        if format[:delimiter].present?
+          sanitized = sanitized.gsub(format[:delimiter], "")
+        end
+      end
+
+      # Replace separator with period for proper float parsing
+      if format[:separator].present?
+        sanitized = sanitized.gsub(format[:separator], ".")
+      end
+
+      # Return empty string if not a valid number
+      unless sanitized =~ /\A-?\d+\.?\d*\z/
+        return ""
+      end
+
+      sanitized
+    end
+
+    def set_default_currency_and_format
```

Comment:
Can we set this to `set_default_number_format` now that we've removed currency?

Reply:
This is done, changed the method name, now we just need to wait for the tests to pass

Reply:
Add another commit since I forgot to change it in `before_validation`